core/templates/domain/topic/editable-topic-backend-api.service.ts

```diff
@@ -16,180 +16,194 @@
+      AppConstants.EDITABLE_TOPIC_DATA_URL_TEMPLATE, {
+        topic_id: topicId
       });
-    };
+    this.http['delete'](topicDataUrl).toPromise().then((response: any) => {
+      if (successCallback) {
+        successCallback(200);
+      }
+    }, (errorResponse) => {
+      if (errorCallback) {
+        errorCallback(errorResponse.body);
+      }
+    });
+  }
 
-    var _updateTopic = function(
-        topicId, topicVersion, commitMessage, changeList,
-        successCallback, errorCallback) {
-      var editableTopicDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_TOPIC_DATA_URL_TEMPLATE, {
-          topic_id: topicId
-        });
+  private _updateTopic(
+      topicId: string, topicVersion: string, commitMessage: string,
+      changeList: Array<Object>,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    let editableTopicDataUrl = this.urlInterpolation.interpolateUrl(
+      AppConstants.EDITABLE_TOPIC_DATA_URL_TEMPLATE, {
+        topic_id: topicId
+      });
 
-      var putData = {
-        version: topicVersion,
-        commit_message: commitMessage,
-        topic_and_subtopic_page_change_dicts: changeList
-      };
-      $http.put(editableTopicDataUrl, putData).then(function(response) {
+    let putData = {
+      version: topicVersion,
+      commit_message: commitMessage,
+      topic_and_subtopic_page_change_dicts: changeList
+    };
+    this.http.put(editableTopicDataUrl, putData).toPromise().then(
+      (response: any) => {
         if (successCallback) {
-          // Here also, a dict with 2 fields are passed instead of just 2
-          // parameters, due to the same reason as written for _fetchTopic().
+        // Here also, a dict with 2 fields are passed instead of just 2
+        // parameters, due to the same reason as written for _fetchTopic().
           successCallback({
-            topicDict: angular.copy(response.data.topic_dict),
-            skillIdToDescriptionDict: angular.copy(
-              response.data.skill_id_to_description_dict),
-            skillIdToRubricsDict: angular.copy(
-              response.data.skill_id_to_rubrics_dict)
+            topicDict: cloneDeep(response.topic_dict),
+            skillIdToDescriptionDict: cloneDeep(
+              response.skill_id_to_description_dict),
+            skillIdToRubricsDict: cloneDeep(response.skill_id_to_rubrics_dict)
           });
         }
-      }, function(errorResponse) {
+      }, (errorResponse) => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(errorResponse.error);
         }
       });
-    };
+  }
 
-    return {
-      fetchTopic: function(topicId) {
-        return $q(function(resolve, reject) {
-          _fetchTopic(topicId, resolve, reject);
-        });
-      },
+  fetchTopic(topicId: string): Promise<object> {
```

Comment:
Promise => Promise. (Notice the capital O)

Reply:
Done